### DISCUSSION TITLE
Future of fetchCount()

### DISCUSSION THREAD

**Discussion by beilCrxmarkets** [querydsl/querydsl#3363] [General]
With the release of version 5.0.0, the method fetchCount (https://github.com/querydsl/querydsl/blob/master/querydsl-jpa/src/main/java/com/querydsl/jpa/impl/AbstractJPAQuery.java) was marked as @Deprecated. 

As far as can be understood, the use of fetchCount for queries with orderBy and having is problematic. Other queries can be calculated without any problems. 

We use fetchCount in a very generic way so we are not able to do without it. In addition, we do not want to add an additional dependency to Blaze. 

We have therefore decided to throw an UnsupportedOpperationException if someone wants to use a query with groupBy or having in combination with fetchCount. Through high test coverage, we could thereby determine that fetchCount is never used in combination with orderBy or having in our platform. 

Our question now is whether it is planned that the method fetchCount will be removed in the future or not?
Furthermore, we wanted to ask for an exception to be thrown in case of calculation in memory. This functionality could be controlled via a parameter.

**Comment by Edward-Li** [comment]
i am also facing this issue, after checking the official doc, it suggests using "Blaze-Persistence integration" for alternative.
as the comment mentioned, it is hard to handle "having" or "order by" etc.

Refer to :
https://querydsl.com/static/querydsl/5.0.0/apidocs/deprecated-list.html
